src/v1.js

```diff
@@ -373,10 +375,25 @@ api.declare({
   var payload = req.body;
   var hook = await this.Hook.load({hookGroupId, hookId}, true);
   var error = null;
+  const ajv = new Ajv();
 
   if (!hook) {
     return res.reportError('ResourceNotFound', 'No such hook', {});
   }
+  //Using ajv lib to check if the context respect the triggerSchema
+  var validate = ajv.compile(hook.triggerSchema);
```

Comment:
Can we throw an exception ?

Reply:
I think it will throw an exception -- but taskcluster-lib-api will turn that into a 500 Internal Server Error.  So we should catch the exception from `ajv.compile` and call `return res.reportError`.

Also, it would be really nice if the user found out about this earlier: when they created or updated the hook, rather than when they tried to trigger the hook.